src/compiler/declarationEmitter.ts

```diff
@@ -1593,7 +1593,10 @@ namespace ts {
                 emitTypeOfVariableDeclarationFromTypeLiteral(node);
             }
             else if (!hasModifier(node.parent, ModifierFlags.Private)) {
-                writeTypeOfDeclaration(node, node.type, getParameterDeclarationTypeVisibilityError);
+                // use the checker's type, not the declared type, for an initialized parameter (that isn't a parameter property)
+                const isInitializedParameter = node.initializer && !(getModifierFlags(node) & ModifierFlags.ParameterPropertyModifier);
+                const typeNode = isInitializedParameter ? undefined : node.type;

```

Comment:
That seems like a good idea. Two notes though:

1. We must not have very much declaration coverage because only two other tests broke.
2. This change makes it clearer than ever that duplicate marking of optionality is the way to go with our current semantics (which ignores the missing-undefined vs provided-undefined distinction).